### DISCUSSION TITLE
cdk actions on multiple accounts using a single IAM role

### DISCUSSION THREAD

**Discussion by yashjain24** [aws/aws-cdk#15835] [Q&A]
I have created an IAM role in account 111111111111 (arn:aws:iam::111111111111:role/cdk-dev-deployment-role) which has an sts:AssumeRole Policy for role in account 222222222222 for cdk actions on account 222222222222 (arn:aws:iam::222222222222:role/cdk-prod-deployment-role).

I am using sts credentials for arn:aws:iam::111111111111:role/cdk-dev-deployment-role

When I try the following actions
cdk diff <StackName222222222222>
cdk diff <StackName222222222222> --role-arn arn:aws:iam::222222222222:role/cdk-prod-deployment-role

Both give the same result "Need to perform AWS calls for account 222222222222, but the current credentials are for 111111111111"

aws-cdk version = 1.83.0

**Comment by peterwoodworth** [comment]
> Need to perform AWS calls for account 222222222222, but the current credentials are for 111111111111

This should print anyway. Are you finding any other behavior/error messages or is this it? 

The modern bootstrapping experience on v2 should make things easier, please see our documentation here https://docs.aws.amazon.com/cdk/v2/guide/bootstrapping.html
